lib.php

```diff
@@ -39,4 +39,55 @@
 define('ACTION_WORKFLOW_INSTANCE_FROM', 'workflow_instance_form');
 define('ACTION_WORKFLOW_DELETE', 'workflow_instance_delete');
 define('ACTION_WORKFLOW_DUPLICATE', 'workflow_instance_duplicate');
-define('ACTION_WORKFLOW_ACTIVATE', 'workflow_instance_activate');
\ No newline at end of file
+define('ACTION_WORKFLOW_ACTIVATE', 'workflow_instance_activate');
+
+/**
+ * Adds a tool_lifecycle link to the course admin menu.
+ *
+ * @param navigation_node $navigation The navigation node to extend
+ * @param stdClass $course The course to object for the tool
+ * @param context $context The context of the course
+ * @return void|null return null if we don't want to display the node.
+ */
+function tool_lifecycle_extend_navigation_course($navigation, $course, $context) {
+    global $PAGE;
+
+    // Only add this settings item on non-site course pages.
+    if (!$PAGE->course || $PAGE->course->id == SITEID) {
+        return null;
+    }
+
+    $url = null;
+    $settingnode = null;
+
+    $url = new moodle_url('/admin/tool/lifecycle/view.php', array(
+        'contextid' => $context->id
```

Comment:
Why do you include the contextid? This is not checked within the view.php. We are just using the system context there, since the view.php is not viewed in the context of a course.